crates/bevy_asset/src/handle.rs

```diff
@@ -58,12 +58,29 @@ impl HandleId {
 ///
 /// Handles contain a unique id that corresponds to a specific asset in the [Assets](crate::Assets)
 /// collection.
+///
+/// # Accessing the Asset
+///
+/// A handle is _not_ the asset itself, but should be seen as a pointer to the asset. Modifying a
+/// handle's `id` only modifies which asset is being pointed to. To get the actual asset, try using
+/// [`Assets::get`](crate::Assets::get) or [`Assets::get_mut`](crate::Assets::get_mut).
+///
+/// # Strong and Weak
+///
+/// A handle can be either "Strong" or "Weak". Simply put: Strong handles keep the asset loaded,
+/// while Weak handles do not affect the loaded status of assets. This is due to a type of
+/// _reference counting_. When the number of Strong handles that exist for any given asset reach
+/// zero, the asset is dropped and becomes unloaded. In some cases, you might want a reference to an
+/// asset but don't want to take the responsibility of keeping it loaded that comes with a Strong handle.
+/// This is where a Weak handle can be very useful.
```

Comment:
I would like to see somewhere an explicit use case where Weak handles are useful rather than some "vague" comment about the responsibility of keeping it loaded.

Not sure if it would be better in doc comments or in an example though.

I use it when I add/remove a lot of components with the same handle in a short time, while keeping the handle in a ressource. Keeping a Strong handle doesn't add costs, it's only when you clone/drop it that extra stuff happens

Reply:
I agree that an example could be useful. Maybe something like this:

> For example, imagine you have a `Sprite` component and a `Collider` component. The `Collider` uses the `Sprite`'s image size to check for collisions. It does so by keeping a Weak copy of the `Sprite`'s Strong handle to the image asset. 
>
> If the `Sprite` is removed, its Strong handle to the image is dropped with it. And since it was the only Strong handle for that asset, the asset is unloaded. Our `Collider` component still has a Weak handle to the unloaded asset, but it will not be able to retrieve the image data, resulting in collisions no longer being detected for that entity.

Does that look correct? I actually haven't used Weak handles all that much, so I could be completely wrong in its use 😅

I could also turn this into a code snippet, but I felt like that might be a bit _too_ verbose. But if a code snippet is easier to follow, I can do that instead/alongside.

Reply:
looks good to me, thanks!